bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -105,23 +122,46 @@ PreProcessor::PreProcessor(shared_ptr<MsgsCommunicator> &msgsCommunicator,
     preProcessResultBuffers_.push_back(Sliver(new char[maxReplyMsgSize_], maxReplyMsgSize_));
   }
   threadPool_.start(numOfClients_);
-  requestsStatusCheckTimer_ = TimersSingleton::getInstance().add(
-      chrono::milliseconds(myReplica_.getReplicaConfig().preExecReqStatusCheckTimerMillisec),
-      Timers::Timer::RECURRING,
-      [this](Timers::Handle h) { onRequestsStatusCheckTimer(h); });
+  RequestProcessingInfo::init(numOfRequiredReplies(), preProcessReqWaitTimeMilli_);
+  addTimers();
 }
 
 PreProcessor::~PreProcessor() {
+  cancelTimers();
+  threadPool_.stop();
+}
+
+void PreProcessor::addTimers() {
+  if (preExecReqStatusCheckTimeMilli_ != 0)
+    requestsStatusCheckTimer_ = TimersSingleton::getInstance().add(
+        chrono::milliseconds(preExecReqStatusCheckTimeMilli_), Timers::Timer::RECURRING, [this](Timers::Handle h) {
+          onRequestsStatusCheckTimer(h);
+        });
```

Comment:
as below, can you also explain what this timer is used for?

Reply:
OK